server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMessagesCreationProcessor.java

```diff
@@ -97,7 +105,8 @@
                                  MailboxId.Factory mailboxIdFactory,
                                  MessageAppender messageAppender,
                                  MessageSender messageSender,
-                                 ReferenceUpdater referenceUpdater) {
+                                 ReferenceUpdater referenceUpdater,
+                                 RecipientRewriteTable recipientRewriteTable) {
```

Comment:
I don't agree we should inject that dependency here: the only thing we want to know here is "does the provided From is allowed for this User". I think that we should build a class for that because it's probably something we will want to share with other parts of the code, that we would like to test in isolation, etc.